### PR TITLE
Xorg Server 21.1.2 Survey (contains security update)

### LINKED ISSUE #3722: xorg-server: security update to 21.1.2
### CVE IDs

CVE-2021-4008,CVE-2021-4009,CVE-2021-4010,CVE-2021-4011

### Other security advisory IDs

- [PDHYZM6FII35JA7J275MFCJO6ADJUPQX](https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PDHYZM6FII35JA7J275MFCJO6ADJUPQX/)
- [T57DCF726O5LLTST4NBL5PQ7DLPB46HT](https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/T57DCF726O5LLTST4NBL5PQ7DLPB46HT/)
- [003122](https://lists.x.org/archives/xorg-announce/2021-December/003122.html)
- [003124](https://lists.x.org/archives/xorg-announce/2021-December/003124.html)
- [DSA-5027](https://www.debian.org/security/2021/dsa-5027)
- [ZDI-21-1547](https://www.zerodayinitiative.com/advisories/ZDI-21-1547/)
- [ZDI-21-1548](https://www.zerodayinitiative.com/advisories/ZDI-21-1548/)
- [ZDI-21-1549](https://www.zerodayinitiative.com/advisories/ZDI-21-1549/)
- [ZDI-21-1550](https://www.zerodayinitiative.com/advisories/ZDI-21-1550/)

### Description

- https://nvd.nist.gov/vuln/detail/CVE-2021-4008
- https://nvd.nist.gov/vuln/detail/CVE-2021-4009
- https://nvd.nist.gov/vuln/detail/CVE-2021-4010
- https://nvd.nist.gov/vuln/detail/CVE-2021-4011

### Patches

N/A

### PoC(s)

N/A

### PR DESCRIPTION
**Pull request by WhiredPlanck**
Topic Description
-----------------

Security update `xorg-server` to 21.1.2, and rebuild `xf86-*`.

Package(s) Affected
-------------------

```
x11-proto
libxcvt
xf86-*
virtualbox
```

Security Update?
----------------

Yes - Issue Number: #3722 

Build Order
-----------

```
1. x11-proto, libxcvt
2. groups/xserver (amd64), groups/xserver.non-amd64 (arm64, loongson3, ppc64el)
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

**Comment by WhiredPlanck** [comment]
LGTM on AMD64

image

File: extra-x11/xf86-input-synaptics/autobuild/beyond [hunk 1]
```diff
@@ -1,2 +1,3 @@
-install -m755 -d "$PKGDIR"/etc/X11/xorg.conf.d
+abinfo 'Copying configurations ...'
+install -dvm755 "$PKGDIR"/etc/X11/xorg.conf.d
```
Comment (MingcongBai) [inline extra-x11/xf86-input-synaptics/autobuild/beyond]: You could have used...

```
install -Dvm644 "$SRCDIR"/conf/*.conf \
    -t "$PKGDIR"/etc/X11/xorg.conf.d/
```

File: extra-x11/xf86-video-intel/autobuild/defines [hunk 2]
```diff
@@ -14,4 +14,10 @@ AUTOTOOLS_AFTER__I486="${AUTOTOOLS_AFTER__RETRO}"
 
 AB_FLAGS_RRO=0
 AB_FLAGS_NOW=0
+
+# Fedora: This package causes LTO to thrash sucking up enormous amounts of VM.
```
Comment (MingcongBai) [inline extra-x11/xf86-video-intel/autobuild/defines:18]: Please test and see if this actually applies to us.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-x11/xf86-video-intel/autobuild/defines (hunk 2 above)
Comment (WhiredPlanck) [reply extra-x11/xf86-video-intel/autobuild/defines:18]: I've test LTO and unfortunately we are also unable to turn it on. So I copied this explanation from Fedora.

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**